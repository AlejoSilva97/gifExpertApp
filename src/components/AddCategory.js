import React, { useState } from 'react'
import PropTypes from 'prop-types';

//Al venir props.setCategories desestructuramos y sacamos directamente {setCategories}
export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('');

    //En e.target.value se guarda el valor actual que tiene el input
    const handleInputChange =(e) =>{
        setinputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        //con este metodo prevenimos que se recargue el navegador al hacer submit
        e.preventDefault();

        if ( inputValue.trim().length > 2 ) {
            //Este metodo setCategories se envia como propiedad desde el elemento padre
            setCategories( cats => [inputValue, ...cats]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit} >
            <p> {inputValue} </p>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
