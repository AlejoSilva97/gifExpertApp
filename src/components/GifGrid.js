import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {

    //Este es un custom hook creado por nosotros mismos
    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
            {/* Estas clases son animaciones de la libreria animate.css */}
            <h3 className="animate__animated animate__fadeIn">{category}</h3>

            {/* Si loading esta true entonces ponemos un parrafo que indica que esta cargando las imagenes */}
            {loading && <p className="animate__animated animate__flash">Loading...</p>}

            <div className="card-grid">
                {
                    images.map((img)=>(
                        <GifGridItem
                            key={img.id} 
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
