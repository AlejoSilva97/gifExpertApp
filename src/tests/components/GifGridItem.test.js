import React from 'react';
import { shallow } from "enzyme";

import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en el componente <GifGridItem />', () => {

    const title = 'Titulo de la imagen';
    const url = 'https://unaimagen.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);

    test('Debe mostrarse el componente <GifGridItem /> correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe tener un parrafo con el title ', () => {

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe(title);
        
    });
    
    test('Debe tener una imagen con url y alt de las props', () => {

        const img = wrapper.find('img');

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
        
    });

    test('el div debe tener animate fadeIn en las clases', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__fadeIn')).toBe(true);
        //expect(div.prop('className')).toContain('animate__fadeIn'); Este lo hize yo y funciona, no se que tan correcto est√© hacerlo de esta manera
        
    });
    
});


