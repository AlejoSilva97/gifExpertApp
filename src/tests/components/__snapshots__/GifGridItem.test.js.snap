// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGridItem /> Debe mostrarse el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo de la imagen"
    src="https://unaimagen.jpg"
  />
  <p>
     
    Titulo de la imagen
     
  </p>
</div>
`;
